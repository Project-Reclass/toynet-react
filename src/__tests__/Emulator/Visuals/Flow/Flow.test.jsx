import React from 'react';

import { renderWithTheme } from "src/common/test-utils/renderWithTheme";
import Flow from "src/Emulator/Visuals/Flow";

const routers = [
  {
    name: 'r1',
    type: 'router',
    connections: [],
  },
  {
   name: 'r2',
    type: 'router',
    connections: [],
  },
];

const switches = [
  {
    name: 's1',
    type: 'switch',
    connections: [],
  },
];

const hosts = [
  {
    name: 'h1',
    type: 'host',
    connections: [],
  },
];

describe('Flow', () => {
  it('should match the snapshot', () => {
    const { container, getByText } = renderWithTheme(<Flow
      switches={switches}
      hosts={hosts}
      routers={routers}
      isTesting={true}
    />);
    expect(getByText(/r1/gi)).toBeInTheDocument();
    expect(getByText(/r2/gi)).toBeInTheDocument();
    expect(getByText(/s1/gi)).toBeInTheDocument();
    expect(getByText(/h1/gi)).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
  it('should be able to render only hosts', () => {
    const { container, getByText } = renderWithTheme(<Flow
      switches={[]}
      routers={[]}
      hosts={hosts}
      isTesting={true}
    />);
    expect(getByText(/h1/gi)).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
  it('should be able to render only switches', () => {
    const { container, getByText } = renderWithTheme(<Flow
      switches={switches}
      routers={[]}
      hosts={[]}
      isTesting={true}
    />);
    expect(getByText(/s1/gi)).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
  it('should be able to render only routers', () => {
    const { container, getByText } = renderWithTheme(<Flow
      switches={[]}
      routers={routers}
      hosts={[]}
      isTesting={true}
    />);
    expect(getByText(/r1/gi)).toBeInTheDocument();
    expect(getByText(/r2/gi)).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});