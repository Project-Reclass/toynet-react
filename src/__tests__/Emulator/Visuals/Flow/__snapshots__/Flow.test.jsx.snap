// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flow should be able to render only hosts 1`] = `
<div>
  <div
    class="react-flow"
  >
    <div
      class="react-flow__renderer react-flow__zoompane"
    >
      <div
        class="react-flow__nodes"
        style="transform: translate(0px,0px) scale(1);"
      >
        <div
          class="react-flow__node react-flow__node-default selectable"
          data-id="h1"
          style="z-index: 3; transform: translate(258px,312px); pointer-events: all; opacity: 0; color: white; background-color: rgb(158, 16, 89);"
        >
          <div
            class="react-flow__handle react-flow__handle-left nodrag target connectable"
            data-handlepos="left"
            data-nodeid="h1"
          />
          <div
            class="css-8pgesn"
          >
            <p
              class="css-1i194x1"
            >
              H1
            </p>
          </div>
          <div
            class="react-flow__handle react-flow__handle-right nodrag source connectable"
            data-handlepos="right"
            data-nodeid="h1"
          />
        </div>
      </div>
      <svg
        class="react-flow__edges"
        height="500"
        width="500"
      >
        <defs>
          <marker
            class="react-flow__arrowhead"
            id="react-flow__arrowclosed"
            markerHeight="12.5"
            markerWidth="12.5"
            orient="auto"
            refX="0"
            refY="0"
            viewBox="-10 -10 20 20"
          >
            <polyline
              fill="#b1b1b7"
              points="-5,-4 0,0 -5,4 -5,-4"
              stroke="#b1b1b7"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1"
            />
          </marker>
          <marker
            class="react-flow__arrowhead"
            id="react-flow__arrow"
            markerHeight="12.5"
            markerWidth="12.5"
            orient="auto"
            refX="0"
            refY="0"
            viewBox="-10 -10 20 20"
          >
            <polyline
              fill="none"
              points="-5,-4 0,0 -5,4"
              stroke="#b1b1b7"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1.5"
            />
          </marker>
        </defs>
        <g
          transform="translate(0,0) scale(1)"
        />
      </svg>
      <div
        class="react-flow__pane"
      />
    </div>
    <div
      class="css-1qjvdnt"
    >
      <button
        class="css-1ypc82u"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="css-yxiis9"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M0,12a1.5,1.5,0,0,0,1.5,1.5h8.75a.25.25,0,0,1,.25.25V22.5a1.5,1.5,0,0,0,3,0V13.75a.25.25,0,0,1,.25-.25H22.5a1.5,1.5,0,0,0,0-3H13.75a.25.25,0,0,1-.25-.25V1.5a1.5,1.5,0,0,0-3,0v8.75a.25.25,0,0,1-.25.25H1.5A1.5,1.5,0,0,0,0,12Z"
            fill="currentColor"
          />
        </svg>
        Host
      </button>
      <button
        class="css-w42zvo"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="css-yxiis9"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M0,12a1.5,1.5,0,0,0,1.5,1.5h8.75a.25.25,0,0,1,.25.25V22.5a1.5,1.5,0,0,0,3,0V13.75a.25.25,0,0,1,.25-.25H22.5a1.5,1.5,0,0,0,0-3H13.75a.25.25,0,0,1-.25-.25V1.5a1.5,1.5,0,0,0-3,0v8.75a.25.25,0,0,1-.25.25H1.5A1.5,1.5,0,0,0,0,12Z"
            fill="currentColor"
          />
        </svg>
        Switch
      </button>
      <button
        class="css-1470nqk"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="css-yxiis9"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M0,12a1.5,1.5,0,0,0,1.5,1.5h8.75a.25.25,0,0,1,.25.25V22.5a1.5,1.5,0,0,0,3,0V13.75a.25.25,0,0,1,.25-.25H22.5a1.5,1.5,0,0,0,0-3H13.75a.25.25,0,0,1-.25-.25V1.5a1.5,1.5,0,0,0-3,0v8.75a.25.25,0,0,1-.25.25H1.5A1.5,1.5,0,0,0,0,12Z"
            fill="currentColor"
          />
        </svg>
        Router
      </button>
    </div>
    <div
      class="react-flow__controls css-1if3jds"
    >
      <div
        class="react-flow__controls-button react-flow__controls-zoomin"
      >
        <svg
          viewBox="0 0 32 32"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M32 18.133H18.133V32h-4.266V18.133H0v-4.266h13.867V0h4.266v13.867H32z"
          />
        </svg>
      </div>
      <div
        class="react-flow__controls-button react-flow__controls-zoomout"
      >
        <svg
          viewBox="0 0 32 5"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h32v4.2H0z"
          />
        </svg>
      </div>
      <div
        class="react-flow__controls-button react-flow__controls-fitview"
      >
        <svg
          viewBox="0 0 32 30"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M3.692 4.63c0-.53.4-.938.939-.938h5.215V0H4.708C2.13 0 0 2.054 0 4.63v5.216h3.692V4.631zM27.354 0h-5.2v3.692h5.17c.53 0 .984.4.984.939v5.215H32V4.631A4.624 4.624 0 0027.354 0zm.954 24.83c0 .532-.4.94-.939.94h-5.215v3.768h5.215c2.577 0 4.631-2.13 4.631-4.707v-5.139h-3.692v5.139zm-23.677.94a.919.919 0 01-.939-.94v-5.138H0v5.139c0 2.577 2.13 4.707 4.708 4.707h5.138V25.77H4.631z"
          />
        </svg>
      </div>
      <div
        class="react-flow__controls-button react-flow__controls-interactive"
      >
        <svg
          viewBox="0 0 25 32"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M21.333 10.667H19.81V7.619C19.81 3.429 16.38 0 12.19 0c-4.114 1.828-1.37 2.133.305 2.438 1.676.305 4.42 2.59 4.42 5.181v3.048H3.047A3.056 3.056 0 000 13.714v15.238A3.056 3.056 0 003.048 32h18.285a3.056 3.056 0 003.048-3.048V13.714a3.056 3.056 0 00-3.048-3.047zM12.19 24.533a3.056 3.056 0 01-3.047-3.047 3.056 3.056 0 013.047-3.048 3.056 3.056 0 013.048 3.048 3.056 3.056 0 01-3.048 3.047z"
          />
        </svg>
      </div>
    </div>
    <svg
      class="react-flow__background"
      style="width: 100%; height: 100%;"
    >
      <pattern
        height="16"
        id="pattern"
        patternUnits="userSpaceOnUse"
        width="16"
        x="0"
        y="0"
      >
        <circle
          cx="0.25"
          cy="0.25"
          fill="#aaa"
          r="0.5"
        />
      </pattern>
      <rect
        fill="url(#pattern)"
        height="100%"
        width="100%"
        x="0"
        y="0"
      />
    </svg>
  </div>
</div>
`;

exports[`Flow should be able to render only routers 1`] = `
<div>
  <div
    class="react-flow"
  >
    <div
      class="react-flow__renderer react-flow__zoompane"
    >
      <div
        class="react-flow__nodes"
        style="transform: translate(0px,0px) scale(1);"
      >
        <div
          class="react-flow__node react-flow__node-default selectable"
          data-id="r1"
          style="z-index: 3; transform: translate(258px,54px); pointer-events: all; opacity: 0; color: white; background-color: rgb(189, 169, 19);"
        >
          <div
            class="react-flow__handle react-flow__handle-left nodrag target connectable"
            data-handlepos="left"
            data-nodeid="r1"
          />
          <div
            class="css-8pgesn"
          >
            <p
              class="css-1i194x1"
            >
              R1
            </p>
          </div>
          <div
            class="react-flow__handle react-flow__handle-right nodrag source connectable"
            data-handlepos="right"
            data-nodeid="r1"
          />
        </div>
        <div
          class="react-flow__node react-flow__node-default selectable"
          data-id="r2"
          style="z-index: 3; transform: translate(258px,140px); pointer-events: all; opacity: 0; color: white; background-color: rgb(189, 169, 19);"
        >
          <div
            class="react-flow__handle react-flow__handle-left nodrag target connectable"
            data-handlepos="left"
            data-nodeid="r2"
          />
          <div
            class="css-8pgesn"
          >
            <p
              class="css-1i194x1"
            >
              R2
            </p>
          </div>
          <div
            class="react-flow__handle react-flow__handle-right nodrag source connectable"
            data-handlepos="right"
            data-nodeid="r2"
          />
        </div>
      </div>
      <svg
        class="react-flow__edges"
        height="500"
        width="500"
      >
        <defs>
          <marker
            class="react-flow__arrowhead"
            id="react-flow__arrowclosed"
            markerHeight="12.5"
            markerWidth="12.5"
            orient="auto"
            refX="0"
            refY="0"
            viewBox="-10 -10 20 20"
          >
            <polyline
              fill="#b1b1b7"
              points="-5,-4 0,0 -5,4 -5,-4"
              stroke="#b1b1b7"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1"
            />
          </marker>
          <marker
            class="react-flow__arrowhead"
            id="react-flow__arrow"
            markerHeight="12.5"
            markerWidth="12.5"
            orient="auto"
            refX="0"
            refY="0"
            viewBox="-10 -10 20 20"
          >
            <polyline
              fill="none"
              points="-5,-4 0,0 -5,4"
              stroke="#b1b1b7"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1.5"
            />
          </marker>
        </defs>
        <g
          transform="translate(0,0) scale(1)"
        />
      </svg>
      <div
        class="react-flow__pane"
      />
    </div>
    <div
      class="css-1qjvdnt"
    >
      <button
        class="css-1ypc82u"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="css-yxiis9"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M0,12a1.5,1.5,0,0,0,1.5,1.5h8.75a.25.25,0,0,1,.25.25V22.5a1.5,1.5,0,0,0,3,0V13.75a.25.25,0,0,1,.25-.25H22.5a1.5,1.5,0,0,0,0-3H13.75a.25.25,0,0,1-.25-.25V1.5a1.5,1.5,0,0,0-3,0v8.75a.25.25,0,0,1-.25.25H1.5A1.5,1.5,0,0,0,0,12Z"
            fill="currentColor"
          />
        </svg>
        Host
      </button>
      <button
        class="css-w42zvo"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="css-yxiis9"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M0,12a1.5,1.5,0,0,0,1.5,1.5h8.75a.25.25,0,0,1,.25.25V22.5a1.5,1.5,0,0,0,3,0V13.75a.25.25,0,0,1,.25-.25H22.5a1.5,1.5,0,0,0,0-3H13.75a.25.25,0,0,1-.25-.25V1.5a1.5,1.5,0,0,0-3,0v8.75a.25.25,0,0,1-.25.25H1.5A1.5,1.5,0,0,0,0,12Z"
            fill="currentColor"
          />
        </svg>
        Switch
      </button>
      <button
        class="css-1470nqk"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="css-yxiis9"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M0,12a1.5,1.5,0,0,0,1.5,1.5h8.75a.25.25,0,0,1,.25.25V22.5a1.5,1.5,0,0,0,3,0V13.75a.25.25,0,0,1,.25-.25H22.5a1.5,1.5,0,0,0,0-3H13.75a.25.25,0,0,1-.25-.25V1.5a1.5,1.5,0,0,0-3,0v8.75a.25.25,0,0,1-.25.25H1.5A1.5,1.5,0,0,0,0,12Z"
            fill="currentColor"
          />
        </svg>
        Router
      </button>
    </div>
    <div
      class="react-flow__controls css-1if3jds"
    >
      <div
        class="react-flow__controls-button react-flow__controls-zoomin"
      >
        <svg
          viewBox="0 0 32 32"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M32 18.133H18.133V32h-4.266V18.133H0v-4.266h13.867V0h4.266v13.867H32z"
          />
        </svg>
      </div>
      <div
        class="react-flow__controls-button react-flow__controls-zoomout"
      >
        <svg
          viewBox="0 0 32 5"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h32v4.2H0z"
          />
        </svg>
      </div>
      <div
        class="react-flow__controls-button react-flow__controls-fitview"
      >
        <svg
          viewBox="0 0 32 30"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M3.692 4.63c0-.53.4-.938.939-.938h5.215V0H4.708C2.13 0 0 2.054 0 4.63v5.216h3.692V4.631zM27.354 0h-5.2v3.692h5.17c.53 0 .984.4.984.939v5.215H32V4.631A4.624 4.624 0 0027.354 0zm.954 24.83c0 .532-.4.94-.939.94h-5.215v3.768h5.215c2.577 0 4.631-2.13 4.631-4.707v-5.139h-3.692v5.139zm-23.677.94a.919.919 0 01-.939-.94v-5.138H0v5.139c0 2.577 2.13 4.707 4.708 4.707h5.138V25.77H4.631z"
          />
        </svg>
      </div>
      <div
        class="react-flow__controls-button react-flow__controls-interactive"
      >
        <svg
          viewBox="0 0 25 32"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M21.333 10.667H19.81V7.619C19.81 3.429 16.38 0 12.19 0c-4.114 1.828-1.37 2.133.305 2.438 1.676.305 4.42 2.59 4.42 5.181v3.048H3.047A3.056 3.056 0 000 13.714v15.238A3.056 3.056 0 003.048 32h18.285a3.056 3.056 0 003.048-3.048V13.714a3.056 3.056 0 00-3.048-3.047zM12.19 24.533a3.056 3.056 0 01-3.047-3.047 3.056 3.056 0 013.047-3.048 3.056 3.056 0 013.048 3.048 3.056 3.056 0 01-3.048 3.047z"
          />
        </svg>
      </div>
    </div>
    <svg
      class="react-flow__background"
      style="width: 100%; height: 100%;"
    >
      <pattern
        height="16"
        id="pattern"
        patternUnits="userSpaceOnUse"
        width="16"
        x="0"
        y="0"
      >
        <circle
          cx="0.25"
          cy="0.25"
          fill="#aaa"
          r="0.5"
        />
      </pattern>
      <rect
        fill="url(#pattern)"
        height="100%"
        width="100%"
        x="0"
        y="0"
      />
    </svg>
  </div>
</div>
`;

exports[`Flow should be able to render only switches 1`] = `
<div>
  <div
    class="react-flow"
  >
    <div
      class="react-flow__renderer react-flow__zoompane"
    >
      <div
        class="react-flow__nodes"
        style="transform: translate(0px,0px) scale(1);"
      >
        <div
          class="react-flow__node react-flow__node-default selectable"
          data-id="s1"
          style="z-index: 3; transform: translate(258px,226px); pointer-events: all; opacity: 0; color: white; background-color: rgb(0, 138, 158);"
        >
          <div
            class="react-flow__handle react-flow__handle-left nodrag target connectable"
            data-handlepos="left"
            data-nodeid="s1"
          />
          <div
            class="css-8pgesn"
          >
            <p
              class="css-1i194x1"
            >
              S1
            </p>
          </div>
          <div
            class="react-flow__handle react-flow__handle-right nodrag source connectable"
            data-handlepos="right"
            data-nodeid="s1"
          />
        </div>
      </div>
      <svg
        class="react-flow__edges"
        height="500"
        width="500"
      >
        <defs>
          <marker
            class="react-flow__arrowhead"
            id="react-flow__arrowclosed"
            markerHeight="12.5"
            markerWidth="12.5"
            orient="auto"
            refX="0"
            refY="0"
            viewBox="-10 -10 20 20"
          >
            <polyline
              fill="#b1b1b7"
              points="-5,-4 0,0 -5,4 -5,-4"
              stroke="#b1b1b7"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1"
            />
          </marker>
          <marker
            class="react-flow__arrowhead"
            id="react-flow__arrow"
            markerHeight="12.5"
            markerWidth="12.5"
            orient="auto"
            refX="0"
            refY="0"
            viewBox="-10 -10 20 20"
          >
            <polyline
              fill="none"
              points="-5,-4 0,0 -5,4"
              stroke="#b1b1b7"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1.5"
            />
          </marker>
        </defs>
        <g
          transform="translate(0,0) scale(1)"
        />
      </svg>
      <div
        class="react-flow__pane"
      />
    </div>
    <div
      class="css-1qjvdnt"
    >
      <button
        class="css-1ypc82u"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="css-yxiis9"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M0,12a1.5,1.5,0,0,0,1.5,1.5h8.75a.25.25,0,0,1,.25.25V22.5a1.5,1.5,0,0,0,3,0V13.75a.25.25,0,0,1,.25-.25H22.5a1.5,1.5,0,0,0,0-3H13.75a.25.25,0,0,1-.25-.25V1.5a1.5,1.5,0,0,0-3,0v8.75a.25.25,0,0,1-.25.25H1.5A1.5,1.5,0,0,0,0,12Z"
            fill="currentColor"
          />
        </svg>
        Host
      </button>
      <button
        class="css-w42zvo"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="css-yxiis9"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M0,12a1.5,1.5,0,0,0,1.5,1.5h8.75a.25.25,0,0,1,.25.25V22.5a1.5,1.5,0,0,0,3,0V13.75a.25.25,0,0,1,.25-.25H22.5a1.5,1.5,0,0,0,0-3H13.75a.25.25,0,0,1-.25-.25V1.5a1.5,1.5,0,0,0-3,0v8.75a.25.25,0,0,1-.25.25H1.5A1.5,1.5,0,0,0,0,12Z"
            fill="currentColor"
          />
        </svg>
        Switch
      </button>
      <button
        class="css-1470nqk"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="css-yxiis9"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M0,12a1.5,1.5,0,0,0,1.5,1.5h8.75a.25.25,0,0,1,.25.25V22.5a1.5,1.5,0,0,0,3,0V13.75a.25.25,0,0,1,.25-.25H22.5a1.5,1.5,0,0,0,0-3H13.75a.25.25,0,0,1-.25-.25V1.5a1.5,1.5,0,0,0-3,0v8.75a.25.25,0,0,1-.25.25H1.5A1.5,1.5,0,0,0,0,12Z"
            fill="currentColor"
          />
        </svg>
        Router
      </button>
    </div>
    <div
      class="react-flow__controls css-1if3jds"
    >
      <div
        class="react-flow__controls-button react-flow__controls-zoomin"
      >
        <svg
          viewBox="0 0 32 32"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M32 18.133H18.133V32h-4.266V18.133H0v-4.266h13.867V0h4.266v13.867H32z"
          />
        </svg>
      </div>
      <div
        class="react-flow__controls-button react-flow__controls-zoomout"
      >
        <svg
          viewBox="0 0 32 5"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h32v4.2H0z"
          />
        </svg>
      </div>
      <div
        class="react-flow__controls-button react-flow__controls-fitview"
      >
        <svg
          viewBox="0 0 32 30"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M3.692 4.63c0-.53.4-.938.939-.938h5.215V0H4.708C2.13 0 0 2.054 0 4.63v5.216h3.692V4.631zM27.354 0h-5.2v3.692h5.17c.53 0 .984.4.984.939v5.215H32V4.631A4.624 4.624 0 0027.354 0zm.954 24.83c0 .532-.4.94-.939.94h-5.215v3.768h5.215c2.577 0 4.631-2.13 4.631-4.707v-5.139h-3.692v5.139zm-23.677.94a.919.919 0 01-.939-.94v-5.138H0v5.139c0 2.577 2.13 4.707 4.708 4.707h5.138V25.77H4.631z"
          />
        </svg>
      </div>
      <div
        class="react-flow__controls-button react-flow__controls-interactive"
      >
        <svg
          viewBox="0 0 25 32"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M21.333 10.667H19.81V7.619C19.81 3.429 16.38 0 12.19 0c-4.114 1.828-1.37 2.133.305 2.438 1.676.305 4.42 2.59 4.42 5.181v3.048H3.047A3.056 3.056 0 000 13.714v15.238A3.056 3.056 0 003.048 32h18.285a3.056 3.056 0 003.048-3.048V13.714a3.056 3.056 0 00-3.048-3.047zM12.19 24.533a3.056 3.056 0 01-3.047-3.047 3.056 3.056 0 013.047-3.048 3.056 3.056 0 013.048 3.048 3.056 3.056 0 01-3.048 3.047z"
          />
        </svg>
      </div>
    </div>
    <svg
      class="react-flow__background"
      style="width: 100%; height: 100%;"
    >
      <pattern
        height="16"
        id="pattern"
        patternUnits="userSpaceOnUse"
        width="16"
        x="0"
        y="0"
      >
        <circle
          cx="0.25"
          cy="0.25"
          fill="#aaa"
          r="0.5"
        />
      </pattern>
      <rect
        fill="url(#pattern)"
        height="100%"
        width="100%"
        x="0"
        y="0"
      />
    </svg>
  </div>
</div>
`;

exports[`Flow should match the snapshot 1`] = `
<div>
  <div
    class="react-flow"
  >
    <div
      class="react-flow__renderer react-flow__zoompane"
    >
      <div
        class="react-flow__nodes"
        style="transform: translate(0px,0px) scale(1);"
      >
        <div
          class="react-flow__node react-flow__node-default selectable"
          data-id="r1"
          style="z-index: 3; transform: translate(258px,54px); pointer-events: all; opacity: 0; color: white; background-color: rgb(189, 169, 19);"
        >
          <div
            class="react-flow__handle react-flow__handle-left nodrag target connectable"
            data-handlepos="left"
            data-nodeid="r1"
          />
          <div
            class="css-8pgesn"
          >
            <p
              class="css-1i194x1"
            >
              R1
            </p>
          </div>
          <div
            class="react-flow__handle react-flow__handle-right nodrag source connectable"
            data-handlepos="right"
            data-nodeid="r1"
          />
        </div>
        <div
          class="react-flow__node react-flow__node-default selectable"
          data-id="r2"
          style="z-index: 3; transform: translate(258px,140px); pointer-events: all; opacity: 0; color: white; background-color: rgb(189, 169, 19);"
        >
          <div
            class="react-flow__handle react-flow__handle-left nodrag target connectable"
            data-handlepos="left"
            data-nodeid="r2"
          />
          <div
            class="css-8pgesn"
          >
            <p
              class="css-1i194x1"
            >
              R2
            </p>
          </div>
          <div
            class="react-flow__handle react-flow__handle-right nodrag source connectable"
            data-handlepos="right"
            data-nodeid="r2"
          />
        </div>
        <div
          class="react-flow__node react-flow__node-default selectable"
          data-id="s1"
          style="z-index: 3; transform: translate(258px,226px); pointer-events: all; opacity: 0; color: white; background-color: rgb(0, 138, 158);"
        >
          <div
            class="react-flow__handle react-flow__handle-left nodrag target connectable"
            data-handlepos="left"
            data-nodeid="s1"
          />
          <div
            class="css-8pgesn"
          >
            <p
              class="css-1i194x1"
            >
              S1
            </p>
          </div>
          <div
            class="react-flow__handle react-flow__handle-right nodrag source connectable"
            data-handlepos="right"
            data-nodeid="s1"
          />
        </div>
        <div
          class="react-flow__node react-flow__node-default selectable"
          data-id="h1"
          style="z-index: 3; transform: translate(258px,312px); pointer-events: all; opacity: 0; color: white; background-color: rgb(158, 16, 89);"
        >
          <div
            class="react-flow__handle react-flow__handle-left nodrag target connectable"
            data-handlepos="left"
            data-nodeid="h1"
          />
          <div
            class="css-8pgesn"
          >
            <p
              class="css-1i194x1"
            >
              H1
            </p>
          </div>
          <div
            class="react-flow__handle react-flow__handle-right nodrag source connectable"
            data-handlepos="right"
            data-nodeid="h1"
          />
        </div>
      </div>
      <svg
        class="react-flow__edges"
        height="500"
        width="500"
      >
        <defs>
          <marker
            class="react-flow__arrowhead"
            id="react-flow__arrowclosed"
            markerHeight="12.5"
            markerWidth="12.5"
            orient="auto"
            refX="0"
            refY="0"
            viewBox="-10 -10 20 20"
          >
            <polyline
              fill="#b1b1b7"
              points="-5,-4 0,0 -5,4 -5,-4"
              stroke="#b1b1b7"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1"
            />
          </marker>
          <marker
            class="react-flow__arrowhead"
            id="react-flow__arrow"
            markerHeight="12.5"
            markerWidth="12.5"
            orient="auto"
            refX="0"
            refY="0"
            viewBox="-10 -10 20 20"
          >
            <polyline
              fill="none"
              points="-5,-4 0,0 -5,4"
              stroke="#b1b1b7"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1.5"
            />
          </marker>
        </defs>
        <g
          transform="translate(0,0) scale(1)"
        />
      </svg>
      <div
        class="react-flow__pane"
      />
    </div>
    <div
      class="css-1qjvdnt"
    >
      <button
        class="css-1ypc82u"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="css-yxiis9"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M0,12a1.5,1.5,0,0,0,1.5,1.5h8.75a.25.25,0,0,1,.25.25V22.5a1.5,1.5,0,0,0,3,0V13.75a.25.25,0,0,1,.25-.25H22.5a1.5,1.5,0,0,0,0-3H13.75a.25.25,0,0,1-.25-.25V1.5a1.5,1.5,0,0,0-3,0v8.75a.25.25,0,0,1-.25.25H1.5A1.5,1.5,0,0,0,0,12Z"
            fill="currentColor"
          />
        </svg>
        Host
      </button>
      <button
        class="css-w42zvo"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="css-yxiis9"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M0,12a1.5,1.5,0,0,0,1.5,1.5h8.75a.25.25,0,0,1,.25.25V22.5a1.5,1.5,0,0,0,3,0V13.75a.25.25,0,0,1,.25-.25H22.5a1.5,1.5,0,0,0,0-3H13.75a.25.25,0,0,1-.25-.25V1.5a1.5,1.5,0,0,0-3,0v8.75a.25.25,0,0,1-.25.25H1.5A1.5,1.5,0,0,0,0,12Z"
            fill="currentColor"
          />
        </svg>
        Switch
      </button>
      <button
        class="css-1470nqk"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="css-yxiis9"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M0,12a1.5,1.5,0,0,0,1.5,1.5h8.75a.25.25,0,0,1,.25.25V22.5a1.5,1.5,0,0,0,3,0V13.75a.25.25,0,0,1,.25-.25H22.5a1.5,1.5,0,0,0,0-3H13.75a.25.25,0,0,1-.25-.25V1.5a1.5,1.5,0,0,0-3,0v8.75a.25.25,0,0,1-.25.25H1.5A1.5,1.5,0,0,0,0,12Z"
            fill="currentColor"
          />
        </svg>
        Router
      </button>
    </div>
    <div
      class="react-flow__controls css-1if3jds"
    >
      <div
        class="react-flow__controls-button react-flow__controls-zoomin"
      >
        <svg
          viewBox="0 0 32 32"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M32 18.133H18.133V32h-4.266V18.133H0v-4.266h13.867V0h4.266v13.867H32z"
          />
        </svg>
      </div>
      <div
        class="react-flow__controls-button react-flow__controls-zoomout"
      >
        <svg
          viewBox="0 0 32 5"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h32v4.2H0z"
          />
        </svg>
      </div>
      <div
        class="react-flow__controls-button react-flow__controls-fitview"
      >
        <svg
          viewBox="0 0 32 30"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M3.692 4.63c0-.53.4-.938.939-.938h5.215V0H4.708C2.13 0 0 2.054 0 4.63v5.216h3.692V4.631zM27.354 0h-5.2v3.692h5.17c.53 0 .984.4.984.939v5.215H32V4.631A4.624 4.624 0 0027.354 0zm.954 24.83c0 .532-.4.94-.939.94h-5.215v3.768h5.215c2.577 0 4.631-2.13 4.631-4.707v-5.139h-3.692v5.139zm-23.677.94a.919.919 0 01-.939-.94v-5.138H0v5.139c0 2.577 2.13 4.707 4.708 4.707h5.138V25.77H4.631z"
          />
        </svg>
      </div>
      <div
        class="react-flow__controls-button react-flow__controls-interactive"
      >
        <svg
          viewBox="0 0 25 32"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M21.333 10.667H19.81V7.619C19.81 3.429 16.38 0 12.19 0c-4.114 1.828-1.37 2.133.305 2.438 1.676.305 4.42 2.59 4.42 5.181v3.048H3.047A3.056 3.056 0 000 13.714v15.238A3.056 3.056 0 003.048 32h18.285a3.056 3.056 0 003.048-3.048V13.714a3.056 3.056 0 00-3.048-3.047zM12.19 24.533a3.056 3.056 0 01-3.047-3.047 3.056 3.056 0 013.047-3.048 3.056 3.056 0 013.048 3.048 3.056 3.056 0 01-3.048 3.047z"
          />
        </svg>
      </div>
    </div>
    <svg
      class="react-flow__background"
      style="width: 100%; height: 100%;"
    >
      <pattern
        height="16"
        id="pattern"
        patternUnits="userSpaceOnUse"
        width="16"
        x="0"
        y="0"
      >
        <circle
          cx="0.25"
          cy="0.25"
          fill="#aaa"
          r="0.5"
        />
      </pattern>
      <rect
        fill="url(#pattern)"
        height="100%"
        width="100%"
        x="0"
        y="0"
      />
    </svg>
  </div>
</div>
`;
